ipaserver/plugins/baseldap.py

```diff
@@ -1937,7 +1937,23 @@ def get_attr_filter(self, ldap, **options):
         """
         search_kw = self.args_options_2_entry(**options)
         search_kw['objectclass'] = self.obj.object_class
-        return ldap.make_filter(search_kw, rules=ldap.MATCH_ALL)
+        default_kw = self.get_default(**options)
+
+        filters = []
+        for name, value in search_kw.items():
+            if name == 'nsaccountlock' and value == False:
+                # If nsaccountlock is False, set the value to True and
+                # search for nsaccountlock != True. This way we can search
+                # for any users where nsaccountlock=False or nsaccountlock=None.
```

Comment:
No magic special cases please, the whole point of doing this in `baseldap` was that it would be generic. If the bit below doesn't work, we need to figure out how to make it work.